acceptance/openstack/compute/v2/servers_test.go

```diff
@@ -11,170 +10,91 @@ import (
-		servers, err := servers.ExtractServers(page)
-		if err != nil {
-			return false, err
-		}
-
-		for _, s := range servers {
-			t.Logf("%s\t%s\t%s\t%s\t%s\t\n", s.ID, s.Name, s.Status, s.AccessIPv4, s.AccessIPv6)
-			count++
-		}
-
-		return true, nil
-	})
-
-	t.Logf("--------\n%d servers listed on %d pages.\n", count, pages)
-}
-
-func networkingClient() (*gophercloud.ServiceClient, error) {
-	opts, err := openstack.AuthOptionsFromEnv()
+	allPages, err := servers.List(client, servers.ListOpts{}).AllPages()
 	if err != nil {
-		return nil, err
+		t.Fatalf("Unable to retrieve servers: %v", err)
 	}
 
-	provider, err := openstack.AuthenticatedClient(opts)
+	allServers, err := servers.ExtractServers(allPages)
 	if err != nil {
-		return nil, err
+		t.Fatalf("Unable to extract servers: %v", err)
 	}
 
-	return openstack.NewNetworkV2(provider, gophercloud.EndpointOpts{
-		Region: os.Getenv("OS_REGION_NAME"),
-	})
+	for _, server := range allServers {
+		printServer(t, &server)
+	}
 }
 
-func createServer(t *testing.T, client *gophercloud.ServiceClient, choices *ComputeChoices) (*servers.Server, error) {
-	if testing.Short() {
-		t.Skip("Skipping test that requires server creation in short mode.")
+func TestServersCreateDestroy(t *testing.T) {
+	client, err := newClient()
+	if err != nil {
+		t.Fatalf("Unable to create a compute client: %v", err)
 	}
 
-	var network networks.Network
-
-	networkingClient, err := networkingClient()
+	choices, err := ComputeChoicesFromEnv()
 	if err != nil {
-		t.Fatalf("Unable to create a networking client: %v", err)
+		t.Fatal(err)
 	}
 
-	pager := networks.List(networkingClient, networks.ListOpts{
-		Name:  choices.NetworkName,
-		Limit: 1,
-	})
-	pager.EachPage(func(page pagination.Page) (bool, error) {
-		networks, err := networks.ExtractNetworks(page)
-		if err != nil {
-			t.Errorf("Failed to extract networks: %v", err)
-			return false, err
-		}
-
-		if len(networks) == 0 {
-			t.Fatalf("No networks to attach to server")
-			return false, err
-		}
-
-		network = networks[0]
-
-		return false, nil
-	})
-
-	name := tools.RandomString("ACPTTEST", 16)
-	t.Logf("Attempting to create server: %s\n", name)
-
-	pwd := tools.MakeNewPassword("")
-
-	server, err := servers.Create(client, servers.CreateOpts{
-		Name:      name,
-		FlavorRef: choices.FlavorID,
-		ImageRef:  choices.ImageID,
-		Networks: []servers.Network{
-			servers.Network{UUID: network.ID},
-		},
-		AdminPass: pwd,
-		Personality: servers.Personality{
-			&servers.File{
-				Path:     "/etc/test",
-				Contents: []byte("hello world"),
-			},
-		},
-	}).Extract()
+	server, err := createServer(t, client, choices)
 	if err != nil {
 		t.Fatalf("Unable to create server: %v", err)
 	}
 
-	th.AssertEquals(t, pwd, server.AdminPass)
-
-	return server, err
-}
+	if err = waitForStatus(client, server, "ACTIVE"); err != nil {
+		t.Fatalf("Unable to wait for server: %v", err)
+	}
+	defer func() {
+		servers.Delete(client, server.ID)
+		t.Logf("Server deleted.")
+	}()
```

Comment:
I've moved the delete defer to be triggered after the instance has been confirmed ACTIVE. This is so if instance creation fails, the error'd instance still remains so the tester can investigate why the instance failed to launch. When defer was in its original location, the instance would immediately be deleted, the UUID never printed (though this can be fixed), and troubleshooting was much harder.